eZ/Bundle/EzPublishCoreBundle/EventListener/ViewControllerListener.php

```diff
@@ -76,42 +94,51 @@ public function getController(FilterControllerEvent $event)
             return;
         }
-        try {
-            if ($request->attributes->has('locationId')) {
-                $valueObject = $this->repository->getLocationService()->loadLocation(
-                    $request->attributes->get('locationId')
-                );
-                $request->attributes->set('contentId', $valueObject->contentId);
-            } elseif ($request->attributes->get('location') instanceof Location) {
-                $valueObject = $request->attributes->get('location');
-                $request->attributes->set('locationId', $valueObject->id);
-                $request->attributes->set('contentId', $valueObject->contentId);
-            } elseif ($request->attributes->has('contentId')) {
-                $valueObject = $this->repository->sudo(
-                    function (Repository $repository) use ($request) {
-                        return $repository->getContentService()->loadContentInfo(
-                            $request->attributes->get('contentId')
-                        );
-                    }
-                );
-            } elseif ($request->attributes->get('contentInfo') instanceof ContentInfo) {
-                $valueObject = $request->attributes->get('contentInfo');
-                $request->attributes->set('contentId', $valueObject->id);
-            }
-        } catch (UnauthorizedException $e) {
-            throw new AccessDeniedException();
+
+        $view = new ContentView(null, $request->attributes->get('viewType'));
+
+        if ($request->attributes->has('locationId')) {
+            $valueObject = $this->repository->getLocationService()->loadLocation(
+                $request->attributes->get('locationId')
+            );
+            $request->attributes->set('contentId', $valueObject->contentId);
+            $request->attributes->set('location', $valueObject);
+
+            $view->setLocation($valueObject);
+        } elseif ($request->attributes->get('location') instanceof Location) {
+            $valueObject = $request->attributes->get('location');
+            $request->attributes->set('locationId', $valueObject->id);
+            $request->attributes->set('contentId', $valueObject->contentId);
+
+            $view->setLocation($valueObject);
         }
 
-        if (!isset($valueObject)) {
-            $this->logger->error('Could not resolver a view controller, invalid value object to match.');
+        if ($request->attributes->has('contentId')) {
+            $contentId = $request->attributes->get('contentId');
+            $content = $this->repository->sudo(
+                function (Repository $repository) use ($contentId) {
+                    return $repository->getContentService()->loadContent($contentId);
+                }
+            );
 
-            return;
+            $location = ($request->attributes->has('location')) ? ($request->attributes->get('location')) : null;
```

Comment:
Can be simplified to:

```php
$location = $request->attributes->get('location');
```

`get()` will return `null` if the attribute is not defined. If you want it to return a default value, just specify it as a 2nd argument.